portfolio/src/main/webapp/script.js

```diff
@@ -95,4 +95,31 @@ function addMarkers(myMap) {
     title: 'Favorite Library',
     icon: iconBase + 'pink-stars.png',
   });
+  addLibraryInfoWindow(myMap, libraryMarker);
+}
+
+/** Generates an info window for a specific marker. */
+function generateInfoWindow(infoString, map, marker) {
+  const infoWindow = new google.maps.InfoWindow({
+    content: infoString,
+  });
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+/** Adds an info window for the library marker. */
+function addLibraryInfoWindow(map, libraryMarker) {
+  const libraryInfoString = '<div id="content">' +
+      '<div id="siteNotice">' +
+      '</div>' +
+      '<h1 id="firstHeading" class="firstHeading">My Favorite Library</h1>' +
+      '<div id="bodyContent">' +
+      '<p>The <b>Chifley Library</b>, named after Joseph Benedict Chifley (Prime Minister of Australia from 1945-1949), ' +
+      'supports the teaching, learning and research activities of staff and students in ANU CASS and CBE.</p>' +
+      '<p>Attribution: ANU Chifley Library, <a href="https://anulib.anu.edu.au/using-library/branches/chifley-library">' +
+      'https://anulib.anu.edu.au/using-library/branches/chifley-library</a></p>' +
+      '</div>' +
+      '</div>';
+  generateInfoWindow(libraryInfoString, map, libraryMarker);
 }
```

Comment:
It seems a little strange to me to include so much HTML in our Javascript. I think we'd prefer to keep these separate where possible, but I don't know enough about HTML/Javascript to know the best practice...

Perhaps you could use the [HTML Content Template element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template)? Or some other HTML template system?

I think it would be good to look into a solution for this, but it probably isn't as important as the test-driven development stuff, so maybe just add a TODO for now?

Reply:
Thank you! Your advice really makes sense. I'm exploring a while about how to convert a local html file into a string, while still having bugs. So, I added TODO now and will keep working on that after I do more for test-driven development.